cachito/workers/pkg_managers/pip.py

```diff
@@ -635,3 +637,347 @@ def _get_variable(self, var_name):
 
         log.error("Variable %r not found along the setup call branch", var_name)
         return None
+
+
+class PipRequirementsFile:
+    """Parse requirements from a pip requirements file."""
+
+    # Comment lines start with optional leading spaces followed by "#"
+    LINE_COMMENT = re.compile(r"(^|\s)+#.*$")
```

Comment:
Do we need the `+` quantifier there? If not need, removing it would make the regex faster.

Reply:
I think the `+` is needed, but could be here instead: `(^|\s+)`. Not sure if that makes any difference

Reference: https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format
> A line that begins with # is treated as a comment and ignored. Whitespace followed by a # causes the # and the remainder of the line to be treated as a comment.

Reply:
I am not sure if I follow the need for the `+` quantifier here. The spec says "**whitespece** followed by a #" and not "**whitespaces** followed by a #". Adding the `+` would make the regex engine match additional spaces for no additional benefits.

Are there any examples where the quantifier would make a difference? Note that this should be used in a `search` and not in a `match` since we are not parsing the spaces from the beginning of the line (^) in that regex.

Reply:
That's a good point. I suggested below to use `sub` instead of `search` or `match`, because for lines that do not start with a comment, we want to ignore only the comment and not the valid part of the line. But that would also not require removing all the extra whitespace, `str.split()` deals with that.

Reply:
pip treats any number of leading white spaces the same:
```
# no leading white spaces
 # one leading white space
         # many leading white spaces
```
So I think `+` is needed. However, it's at the wrong place on my regex. It should be `"^\s*#"`

Reply:
No, I'm wrong. See the other thread about `search` vs `match`

Reply:
This is now handled properly, let me know what you think.